src/main/scala/exercise2/QueueLike.scala

```diff
@@ -0,0 +1,45 @@
+package exercise2
+
+import scala.util.Try
+
+trait QueueLike[T] {
+  val inQueue: StackLike[T]
```

Comment:
I wouldn't define those here. this implies that every queue like collection should have an input stack and an output stack, which is not necessarily true - one could also implement a queue in another way.